SRTO_FC socket option: minimum allowed value is 32 (#1921)
Haivision/srt: docs/API/API-socket-options.md

| ----------------- | ----- | -------- | --------- | ------ | -------- | ------ | --- | ------ |
| `SRTO_FC`         |       | pre      | `int32_t` | pkts   | 25600    | 32..   | RW  | GSD    |

> Flow Control limits the maximum number of packets "in flight" - payload (DATA) packets that were sent

Reviewer: Why is "DATA" all caps?

Author: Just some bad habit. Corrected.

but reception is not yet acknowledged with an ACK control packet.

> It also includes DATA packets already received, but can't be acknowledged due to loss of preceding DATA packet(s).

Reviewer: ```suggestion
It also includes DATA packets already received, but that can't be acknowledged due to loss of preceding DATA packet(s).
```

> It also includes DATA packets already received, but can't be acknowledged due to loss of preceeding DATA packet(s).

Reviewer 2: ```suggestion
It also includes DATA packets already received, but can't be acknowledged due to loss of preceding DATA packet(s).
```

> In other words, if DATA packet with sequence number `A` was lost, then acknowledging of the following `SRTO_FC` packets

Reviewer 2: ```suggestion
In other words, if DATA packet with sequence number `A` was lost, then acknowledgement of the following `SRTO_FC` packets
```

> In other words, if DATA packet with sequence number `A` was lost, then acknowledgement of the following `SRTO_FC` packets

Reviewer 2: ```suggestion
In other words, if a DATA packet with sequence number `A` was lost, then acknowledgement of the following `SRTO_FC` packets
```

is blocked until packet `A` is either successfully retransmitted or dropped by the
[Too-Late Packet Drop mechanism](https://datatracker.ietf.org/doc/html/draft-sharabayko-srt-00#section-4.6).
Thus the sender will have  `SRTO_FC` packets in flight, and will not be allowed to send further DATA packets.

> Therefore `SRTO_FC` is recommended to include possible delays due to packet loss and reransmission.

Reviewer 2: ```suggestion
Therefore, it is recommended `SRTO_FC` include possible delays due to packet loss and retransmission.
```

> Therefore, it is recommended `SRTO_FC` include possible delays due to packet loss and retransmission.

Reviewer 2: ```suggestion
Therefore, when establishing the value of `SRTO_FC`, we recommend taking into consideration possible delays due to packet loss and retransmission.
```

There is a restriction that the receiver buffer size ([SRTO_RCVBUF](#SRTO_RCVBUF)) must not be greater than `SRTO_FC`
([#700](https://github.com/Haivision/srt/issues/700)).
Therefore, it is recommended to set the value of `SRTO_FC` first, and then the value of `SRTO_RCVBUF`.

The default flow control window size is 25600 packets. It is approximately:

> - 270 Mbits of payload in the defaultlive streaming configuration with 1316 bytes SRT payload size;

Reviewer: ```suggestion
- 270 Mbits of payload in the default live streaming configuration with an SRT payload size of 1316 bytes;
```

> - 300 Mbits of payload with 1456 bytes SRT payload size.

Reviewer: ```suggestion
- 300 Mbits of payload with an SRT payload size of 1456 bytes.
```

The minimum number of packets in flight should be (assuming max payload size):  
FCmin = bps / 8 × RTTsec / (MSS - 44),  
where
- `bps` - is the payload bitrate of the stream in bits per second;
- `RTTsec` - RTT of the network connection in seconds;
- `MSS` - Maximum segment size (aka MTU), see [SRTO_MSS](#SRTO_MSS);

> - 44 - size of headers (20 bytes IPv4 + 8 bytes of UDP + 16 bytes of SRT packet header)..

Reviewer: ```suggestion
- 44 - size of headers (20 bytes IPv4 + 8 bytes of UDP + 16 bytes of SRT packet header).
```
